### PR TITLE
Fixes in TOF  matching for multi-TF processing

### PR DESCRIPTION
**Pull request by shahor02**

### CODE REVIEW

File: Detectors/GlobalTrackingWorkflow/tofworkflow/src/RecoWorkflowSpec.cxx [hunk 1]
```diff
@@ -139,9 +135,6 @@ class TOFDPLRecoWorkflowTask
     pc.outputs().snapshot(Output{o2::header::gDataOriginTOF, "CALIBINFOS", 0, Lifetime::Timeframe}, mMatcher.getCalibVector());
 
     // declare done
-    finished = true;
-    //pc.services().get<ControlService>().readyToQuit(QuitRequest::Me);
-    pc.services().get<ControlService>().endOfStream();
```
Comment (sawenzel) [inline Detectors/GlobalTrackingWorkflow/tofworkflow/src/RecoWorkflowSpec.cxx:144]: so how is this processor shutting itself down?

**Comment by shahor02** [comment]
the processors don't need to do this, the end of processing is signaled by upstream devices whose input they depend on.

**Comment by shahor02** [comment]
@noferini is this PR ok for you, can I merge it?

**Comment by shahor02** [comment]
@noferini can you please check this?

File: Detectors/GlobalTrackingWorkflow/tofworkflow/src/RecoWorkflowSpec.cxx (hunk 1 above)
Comment (noferini) [reply Detectors/GlobalTrackingWorkflow/tofworkflow/src/RecoWorkflowSpec.cxx:144]: It should stop when the previous processors (the ones providing input for it) send their endOfStream. Am I wrong? Anyway, it seems to work in this way.

**Pull request merged**